### PR TITLE
Prepare 0.15.0 release

### PR DESCRIPTION
**Pull request by mtreinish**
This commit prepares the 0.15.0 release. It moves the release notes for the release into a separate directory and adds a prelude to the release notes. This should be the last commit to merge for the 0.15.0 release and after it merges that commit will be what is should be tagged as the 0.15.0 release.

As part of this there are few small changes other changes being made. The release notes are update to have consistent formatting and wording.

TODO:

- [x] Wait for last milestone PRs to merge

### CODE REVIEW

**Comment by mtreinish** [comment]
All the tracked PRs for 0.15.0 have merged so this should be unblocked now. Once this merges we can tag the 0.15.0 release.

**Comment by IvanIsCoding** [comment]
I’ll review with more details but if I am not mistaken, we never added release notes for #1213

**Comment by mtreinish** [comment]
Oh ok, I can add one for that

File: releasenotes/notes/0.15/dorogovtsev-goltsev-mendes-generator-d7fd575b693bd652.yaml [hunk 1]
```diff
@@ -3,6 +3,15 @@ features:
...
+    .. jupyter-execute::
+
+      import rustworkx as rx
+      from rustworkx.visualization import mpl_draw
+
+      graph = rx.generators.dorogovtsev_goltsev_mendes_graph(2)
+      mpl_draw(graph)
+
   - |
     Added to rustworkx-for ``generators::dorogovtsev_goltsev_mendes_graph`` function that generates,
```
Comment (raynelfss) [inline releasenotes/notes/0.15/dorogovtsev-goltsev-mendes-generator-d7fd575b693bd652.yaml:16]:
```suggestion
    Added to rustworkx-core ``generators::dorogovtsev_goltsev_mendes_graph`` function that generates,

File: releasenotes/notes/0.15/fix-panic-3962ad36788cab00.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ fixes:
       * :func:`rustworkx.bellman_ford_shortest_paths`
       * :func:`rustworkx.astar_shortest_path`
 
-    where a `Pyo3.PanicException`were raise with no much detail at the moment
-    of pass in the `source` argument the index of an out of bound node.
+    where a ``PanicException`` was raised without much detail when an
+    invalid node index was passed in to the ``source`` argument. This has been
+    corrected so a ``IndexError`` is raised instead.
```
Comment (raynelfss) [inline releasenotes/notes/0.15/fix-panic-3962ad36788cab00.yaml:14]:
```suggestion
    corrected so an ``IndexError`` is raised instead.
```

**Review by raynelfss — APPROVED** [review]
LGTM, Just a couple of minor typos. And also I believe that the release note for #1167 is not present here?

File: releasenotes/notes/0.15/swap-nox-tox-dea2bb14c400641c.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+other:
+  - |
+    For developement of rustworkx the automated automated testing environment
```
Comment (IvanIsCoding) [inline releasenotes/notes/0.15/swap-nox-tox-dea2bb14c400641c.yaml:4]:
```suggestion
    For developement of rustworkx the automated testing environment
```

File: releasenotes/notes/0.15/prepare-0.15.0-771047bc5b71c569.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+---
+prelude: >
+  This is a new feature release of Rustworkx that adds a many new features to
```
Comment (IvanIsCoding) [inline releasenotes/notes/0.15/prepare-0.15.0-771047bc5b71c569.yaml:3]:
```suggestion
  This is a new feature release of Rustworkx that adds many new features to
```

**Review by IvanIsCoding — APPROVED** [review]

**Pull request merged**